app/ide-desktop/lib/dashboard/src/layouts/AssetSearchBar.tsx

```diff
@@ -193,7 +197,7 @@ export default function AssetSearchBar(props: AssetSearchBarProps) {
       document.removeEventListener('keydown', onKeyDown)
       document.removeEventListener('keyup', onKeyUp)
     }
-  }, [setQuery])
+  }, [setQuery, /* should never change */ modalRef])
```

Comment:
I'm not sure if I understand this comment, but it may be just because I'm not familiar enough with the codebase.

Reply:
yeah the comment is a little awkward, not sure how useful it is in practice. the array here is a dependency list, so the function is re-run when any of the dependencies change. the comment indicates that it is likely a mistake if th value changes - it is required to be in the dependency list because of our linter setup.